be/src/exec/pipeline/pipeline_driver_executor.cpp

```diff
@@ -283,8 +283,9 @@ void GlobalDriverExecutor::cancel(DriverRawPtr driver) {
 void GlobalDriverExecutor::report_exec_state(QueryContext* query_ctx, FragmentContext* fragment_ctx,
                                              const Status& status, bool done, bool attach_profile) {
     auto* profile = fragment_ctx->runtime_state()->runtime_profile();
+    std::shared_ptr<ObjectPool> obj_pool = std::make_shared<ObjectPool>();
```

Comment:
no need to use shared_ptr, unique_ptr is enough, because this obj_pool won't be shared by many owners

Reply:
It will be captured by the lambda down below, and the ownership will be transformed to the lambda.